File: Solutions/1394-LargestLuckyInt/LargestLuckyInt.ts
function findLucky(arr: number[]): number {
  let freqMap = new Map<number, number>();

  for (let num of arr) {
    freqMap.set(num, (freqMap.get(num) || 0) + 1);
  }

  let largestLuckyInt = -1; // Default if no lucky int found
  for (const [num, freq] of freqMap) {
    if (num === freq) {
      largestLuckyInt = Math.max(largestLuckyInt, num);
    }
  }

  return largestLuckyInt;
}

File: Solutions/All/1394-LargestLuckyInt/LargestLuckyInt.go
package solution

func findLucky(arr []int) int {
	freqMap := make(map[int]int)

	for _, num := range arr {
		freqMap[num] = freqMap[num] + 1
	}

	largestLuckyInt := -1
	for num, freq := range freqMap {
		if num == freq {
			if num > largestLuckyInt {
				largestLuckyInt = num
			}
		}
	}

	return largestLuckyInt
}
